### ISSUE TITLE
 Cython modules that use __init__.pyx instead of __init__.py are broken with sphinxcontrib.apidoc

### ISSUE THREAD

**Issue by ghost** [sphinx-doc/sphinx#6327]
**Describe the bug**
Cython modules that use `__init__.pyx`/`__init__.so` instead of `__init__.py` are broken with `sphinxcontrib.apidoc`. They appear to be not picked up correctly by whatever internal component that uses (apidoc or autodoc I assume) and it gives a strange listing of the *parent* folder of the module only

**To Reproduce**
```
$ git clone https://github.com/wobblui/wobblui ./testblah
$ cd testblah/docs
$ make html
$ # open ./build/html/reference/modules.html and see it lists the "src" folder instead of the "wobblui" module
```

**Expected behavior**
`docs/build/html/reference/modules.html ` will list "wobblui" module after docs build (that DOES happen by the way if you `touch src/wobblui/__init__.py`, but of course that breaks the module and the functions in `src/wobblui/__init__.pyx` may not be documented properly)

**Your project**
https://github.com/wobblui/wobblui (docs are inside "docs" folder, try "make html" as in steps above)

**Screenshots**
![brokendocs](https://user-images.githubusercontent.com/1573352/56982302-4a1eda80-6b81-11e9-83a9-807710b60438.png)

**Environment info**
- OS: Linux
- Python version: 3.7.3
- Sphinx version: `sphinx-quickstart --version` says 1.7.6
- Sphinx extensions:  sphinxcontrib.apidoc should be the relevant one
- Extra tools: none that I'm aware of(?)

**Additional context**
I already filed a ticket here: https://github.com/sphinx-contrib/apidoc/issues/13 and it was suggested that the problem is probably in sphinx itself or apidoc/autodoc

**Comment by ghost** [comment]
Is anyone aware of a workaround? Other than just adding an `__init__.py` file which doesn't fully do the job since then the `__init__.pyx` won't get documented (but at least the other modules will I suppose). I would also be interested in trying out any dev version of sphinx with an experimental change to improve this

**Comment by tk0miya** [comment]
Unfortunately, current implementation of apidoc only refers `__init__.py`. So we need to modify apidoc to do that. But I'm not familiar with apidoc. So it is a bit tough work to me...

**Comment by tk0miya** [comment]
Fixed by #7113 
Thank you for reporting
